WordPress/Classes/ViewRelated/Plugins/PluginListViewModel.swift

```diff
@@ -3,17 +3,49 @@ import WordPressFlux
 
 protocol PluginPresenter: class {
     func present(plugin: Plugin, capabilities: SitePluginCapabilities)
+    func present(directoryEntry: PluginDirectoryEntry)
 }
 
 class PluginListViewModel: Observable {
+    enum PluginResults: Equatable {
+        case installed(Plugins)
+        case directory([PluginDirectoryEntry])
+
+        init?(_ plugins: Plugins?) {
```

Comment:
I don't think it's very useful to have an optional initializer that depends on an optional argument, since you can avoid calling it if `plugins` is `nil` (see [style guide](https://github.com/wordpress-mobile/swift-style-guide/blob/master/README.md#deal-with-optionals-early))

Instead of calling it like this:

```swift
return PluginResults(store.getPlugins(site: site))
```

you can just do a `map`:

```swift
return store.getPlugins(site: site).map(PluginResults.init)
```